File: LeetCode75_LevelTwo/Longest_Common_Prefix/Longest_Common_Prefix.py
# 14. Longest Common Prefix Python

class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if not strs: return ""
        for index in range(1, len(strs)):
            s = strs[index]
            if len(s) < len(prefix):
                prefix = prefix[:len(s)]
            for i in range(len(s)):
                if i == len(prefix): break
                if s[i] != prefix[i]:
                    prefix = prefix[:i]
                    break
        return prefix

File: Top_Interview_150/C++/Longest Common Prefix.cc
// 14. Longest Common Prefix
#include <vector>
#include <string>
#include <algorithm>

using namespace std;
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int length = strs.size();
        if (length == 0) return "";

        string prefix = "";
        sort(strs.begin(), strs.end());
        string a = strs[0];
        string b = strs[length-1];
        for (int i = 0; i < a.size(); i++) {
            if (a[i] == b[i]){
                prefix += a[i];
            } else break;
        }
        return prefix;
    }
};
